pkg/backporter/httpcache_test.go

```diff
@@ -0,0 +1,180 @@
+package backporter
+
+import (
+	"bytes"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"net/http/httputil"
+	"reflect"
+	"testing"
+)
+
+//bugzillaCache tests
+func TestBugzillaCacheSet(t *testing.T) {
+	testcases := []struct {
+		name         string
+		cache        map[string][]byte
+		key          string
+		value        []byte
+		postSetCache map[string][]byte
+	}{
+		{
+			name:         "Entry is already present",
+			cache:        map[string][]byte{"key": []byte("Test string")},
+			key:          "key",
+			value:        []byte("New data"),
+			postSetCache: map[string][]byte{"key": []byte("New data")},
+		},
+		{
+			name:         "New entry",
+			cache:        map[string][]byte{},
+			key:          "key",
+			value:        []byte("New data"),
+			postSetCache: map[string][]byte{"key": []byte("New data")},
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			bc := newBugzillaCache()
+			bc.cache = tc.cache
+			bc.set(tc.key, tc.value)
+			if !reflect.DeepEqual(tc.postSetCache, bc.cache) {
+				t.Errorf("cached entry mismatch")
```

Comment:
We usually do this kind of check by producing a diff and verifying that the diff is empty, the current approach makes it impossible to debug without a debugger: https://github.com/openshift/ci-tools/blob/196398e6cfff28d99285ff22fab979b243ff6e60/pkg/defaults/defaults_test.go#L423